### ISSUE TITLE
Release process needs to be documented better.

### ISSUE THREAD

**Issue by RyanZim** [istanbuljs/nyc#894]
According to https://www.npmjs.com/package/nyc, the latest version is 12.0.2, but the latest version here is 13.0.0. Why isn't v13 on npm?

**Comment by welkie** [comment]
Just ran into this today. Told my team mate that v13 was avail, and his npm install command gets him v12.

It looks like it is the same version according to the publish dates, just that the version numbers are out of sync. v13 on GitHub, v12 on NPM.

**Comment by coreyfarrell** [comment]
v13 is on npm but it's not yet the 'latest' version.  Currently it can be installed using `npm install nyc@next`.

Although this is not a bug I'm leaving this open until v13 is upgraded to latest, that way other people can see why.  Probably some time in the next week v13 will become latest.

**Comment by RyanZim** [comment]
Ah, OK; understood. Is `next` normally used for releasing `nyc` versions? If so, how about documenting this at the top of the CHANGELOG?

**Comment by welkie** [comment]
It sounds to me like nyc uses a relatively conservative release schedule. Once a new major version is released, it doesn't gain the `latest` tag right away, to allow time for it to be tested first by early adopters? If that's what's going on, I actually like that approach.

**Comment by RyanZim** [comment]
Sure, I'm perfectly fine with the release system, I just think it should be documented someplace.

**Comment by welkie** [comment]
@bcoe Can you confirm this is the case? And if so, would you take a PR for documenting this?

**Comment by coreyfarrell** [comment]
@RyanZim we do release to `next` first before releasing to `latest`.  NYC v13 has been in `next` status for longer than normal as it is the first version to use Babel 7 (upgraded from babel 6).

@welkie Please hold off on that PR for now.  When I get a chance I plan on looking into what it would take to use semver prerelease versions for `next` releases in the future (ie 14.0.0-beta.1).  @bcoe has some concerns about how this will affect the change logs so I need to experiment with this.  In any case I don't want you to spend time working on documenting the current process when it might be about to change.

**Comment by welkie** [comment]
@coreyfarrell Sounds good, thanks for the update.

**Comment by RyanZim** [comment]
@coreyfarrell Any updates?

**Comment by coreyfarrell** [comment]
@welkie sorry for lack of response, life got busy and I dropped the ball on this.  I'd be glad to have a documentation PR.  Maybe a section on about `Testing versions` towards the end of the readme?  The changelog is generated from the commit log so I don't think we can document it there.

The basic process:
1. We publish everything from https://github.com/istanbuljs/istanbuljs/ to npm under the `next` tag, then publish `nyc` to it's `next` tag as well.
2. We give some amount of time depending on the risk level of the updates, the first release of v13 was a special case as we upgraded from babel 6 to babel 7.  Normally we do some testing against other major/large projects.
3. If we're happy with the results we promote the packages on npm to `latest`.

In all likelihood we will soon be releasing nyc@14.0.0.  At first this will be installed using `npm install -D nyc@next`.  Once we're happy that the new release doesn't cause any major regressions we will make it an official release.

I've just created a label [New Release](https://github.com/istanbuljs/nyc/labels/New%20Release) which I will tag the issue I post calling for testing, that link should work for a stable way to query 'is a release pending'.
